homeassistant/components/overkiz/config_flow.py

```diff
@@ -67,7 +72,27 @@ async def async_step_user(
                 errors["base"] = "unknown"
                 LOGGER.exception(exception)
             else:
+                if self._config_entry:
```

Comment:
We need to prevent them from changing accounts in the reauth process otherwise the unique id could be incorrect.

Reply:
Thinking about how to fix this, since the unique id is linked to a gateway (in an account) and not to the email address. And it could be possible to change the emailaddress linked to an account.

Reply:
If the unique id changes we need to reject the credentials and let them re-enter the correct ones

Reply:
@bdraco are there helpers for this? (similar to `_abort_if_unique_id_configured` but the other way around). I cannot seem to find them, thus I would need to write my own compare mechanism?

Reply:
Added a proposal in d986fb9b65b791d2586a4945f2824a700fce116d. Will add tests if this is the way to go.

<img width="461" alt="image" src="https://user-images.githubusercontent.com/1424596/150973249-df24edd2-371c-42cc-be1b-fa24c60f80fb.png">

Reply:
LGTM

Reply:
Added test in c6f97d1ce15b895289c7d7352557714668170fce.